lib/chef/knife/bootstrap.rb

```diff
@@ -308,12 +321,17 @@ def bootstrap_context
       end
 
       def render_template
+        @config[:first_boot_attributes].merge!(@config[:first_boot_attributes_from_file]) if @config[:first_boot_attributes_from_file]
```

Comment:
what would be better here would be to extract an accessor:

```ruby
def first_boot_attributes
   @config[:first_boot_attributes].any? ? @config[:first_boot_attributes] : @config[:first_boot_attributes_from_file]
end
```

If you can make @config[:first_boot_attributes] default to nil if its not passed then you could simplify this to:

```ruby
def first_boot_attributes
   @config[:first_boot_attributes] || @config[:first_boot_attributes_from_file] || {}
end
```

Which would be much more readable.